datastore.go

```diff
@@ -79,15 +79,25 @@ var DefaultOptions Options
 
 func init() {
 	DefaultOptions = Options{
-		GcDiscardRatio: 0.2,
+		GcDiscardRatio: 0.01,
 		GcInterval:     15 * time.Minute,
 		GcSleep:        10 * time.Second,
 		Options:        badger.LSMOnlyOptions(""),
 	}
-	// This is to optimize the database on close so it can be opened
-	// read-only and efficiently queried. We don't do that and hanging on
-	// stop isn't nice.
-	DefaultOptions.Options.CompactL0OnClose = false
+	// This is to optimize the database on closure
+	DefaultOptions.Options.CompactL0OnClose = true
+	
+	// Remove elements which the has been deleted from the database
+	DefaultOptions.Options.NumVersionsToKeep = 0
+	
+	// Reduce the number of zero tables (which are hold in memory)
+	DefaultOptions.Options.NumLevelZeroTables = 1
```

Comment:
This will help compact tables faster, but is likely too aggressive and probably not a huge issue. Tables are 16MiB each so the default of 5 will cause compaction to kick in when we hit 80MiB.

Reply:
The idea was to reduce memory consumption and do the compaction more often, than in larger intervals.

Sure it will hurt the performance. But I didn't want to change the settings, just do a PR for @jsign since he seems to be busy.

Reply:
@RubenKelevra, @Stebalien, please read what I originally wrote with precision [here](https://github.com/ipfs/go-ds-badger/issues/54#issuecomment-613418897), in particular:
> Right tunning of these params might *require some extra work* to discover the perf cost and right tradeoff.

So, my comment was never meant to be directly applied by any means nor asked a PR to be created. What I commented was validated by [this](https://github.com/dgraph-io/badger/issues/1297) and a small program that shows how those knobs effect sizes. (I see now that Ibrhaim _seems_ gave some wrong advice)

So, my goal was to help shed more light on how someone using this repo could have control on this long-problem. 

> But I didn't want to change the settings, just do a PR for @jsign since he seems to be busy.

@RubenKelevra, I've never asked you to do any PR for me. The ones that I know are ready to do, be sure I can do it myself (maybe that's why I might be busy ;))
What I did was invest some time in giving more light about a long-haul problem this repo had, just to help other people using this repo who to possibly deal with it. Jumping to say that was a definitive solution is wrong as I quoted myself above.